plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystem.java

```diff
@@ -505,6 +506,120 @@ public void testFileIteratorListing()
+        //      part=nested/
+        //          parent/
+        //             _nested-hidden-file.txt
+        //             nested-file.txt
+        //      part=plus+sign/
+        //          plus-file.txt
+        //      part=level1|level2/
+        //          pipe-file.txt
+        //          parent1/
+        //             parent2/
+        //                nested-file.txt
+        //      part=level1 | level2/
+        //          pipe-blanks-file.txt
+        //      empty-directory/
+        //      .hidden-in-base.txt
+        Path basePath = new Path(getBasePath(), "test-file-iterator-partitioned-listing");
+        FileSystem fs = hdfsEnvironment.getFileSystem(TESTING_CONTEXT, basePath);
+        TrinoFileSystem trinoFileSystem = new HdfsFileSystemFactory(hdfsEnvironment, new TrinoHdfsFileSystemStats()).create(SESSION);
+        fs.mkdirs(basePath);
+
+        // create file in hidden folder
+        Path fileInHiddenParent = new Path(new Path(basePath, ".hidden"), "nested-file-in-hidden.txt");
+        fs.createNewFile(fileInHiddenParent);
+        // create hidden file in non-hidden folder
+        Path hiddenFileUnderPartitionSimple = new Path(new Path(basePath, "part=simple"), "_hidden-file.txt");
+        fs.createNewFile(hiddenFileUnderPartitionSimple);
+        // create file in `part=simple` non-hidden folder
+        Path plainFilePartitionSimple = new Path(new Path(basePath, "part=simple"), "plain-file.txt");
+        fs.createNewFile(plainFilePartitionSimple);
+        Path nestedFilePartitionNested = new Path(new Path(new Path(basePath, "part=nested"), "parent"), "nested-file.txt");
+        fs.createNewFile(nestedFilePartitionNested);
+        // create hidden file in non-hidden folder
+        Path nestedHiddenFilePartitionNested = new Path(new Path(new Path(basePath, "part=nested"), "parent"), "_nested-hidden-file.txt");
+        fs.createNewFile(nestedHiddenFilePartitionNested);
+        // create file in `part=plus+sign` non-hidden folder (which contains `+` special character)
+        Path plainFilePartitionPlusSign = new Path(new Path(basePath, "part=plus+sign"), "plus-file.txt");
+        fs.createNewFile(plainFilePartitionPlusSign);
+        // create file in `part=level1|level2` non-hidden folder (which contains `|` special character)
+        Path plainFilePartitionPipeSign = new Path(new Path(basePath, "part=level1|level2"), "pipe-file.txt");
+        fs.createNewFile(plainFilePartitionPipeSign);
+        Path deeplyNestedFilePartitionPipeSign = new Path(new Path(new Path(new Path(basePath, "part=level1|level2"), "parent1"), "parent2"), "deeply-nested-file.txt");
+        fs.createNewFile(deeplyNestedFilePartitionPipeSign);
+        // create file in `part=level1 | level2` non-hidden folder (which contains `|` and blank space special characters)
+        Path plainFilePartitionPipeSignBlanks = new Path(new Path(basePath, "part=level1 | level2"), "pipe-blanks-file.txt");
```

Comment:
hive connector would replace spaces with `%20`, right?

Reply:
> No change seems to be occuring.

right, space is not on the list of `io.trino.plugin.hive.util.HiveUtil#PATH_CHAR_TO_ESCAPE`

we probably should add a test case with a character that is
for example the partition path may contain `%` (legal path character)
but it can also contain `%20` (url encoded space). no decoding should occur (https://github.com/trinodb/trino/pull/18167)

can be a follow-up